src/Clapi/Relay.hs

```diff
@@ -137,3 +139,19 @@ handleOwnerMessages vs msgs = (rmsgs, rvs)
     (vErrs, vvs) = vsValidate vs'
     errMsgs = map (\(p, es) -> MsgError p (T.pack es)) (Map.assocs errs)
     rmsgs = if errored then errMsgs else msgs  -- FIXME: success msgs should be derived from tree diff
+
+handleUnclaimedMessages :: Valuespace Validated -> [Message] -> ([Message], Valuespace Validated)
+handleUnclaimedMessages vs msgs = if isValidClaim then handleOwnerMessages vs msgs else (errorAll, vs)
+  where
+    isValidClaim = errorStr == ""
+    claims = filter isClaim msgs
+    isClaim m = case m of
+        (MsgAssignType p _) -> length p == 1
```

Comment:
There's an `isNamespace` in `src/Clapi/NamespaceTracker.hs:37-39`. It likely won't change, but it might be more readable than this code. Using it might require you to put it somewhere common, like `src/Clapi/Path.hs`.

Reply:
Fell out in refactoring but good to know.